### PR TITLE
Add the size param to Recaptcha plugin

### PR DESCRIPTION
**Pull request by g3offrey**
## Description

This PR enable users to set a size to the Recaptcha 2.0 plugin (compact or normal).
As you can see on the google documentation : https://developers.google.com/recaptcha/docs/display
## Example
- Before (the field was in normal size): 

registrationformbefore
- After (the field **can** be in compact size):

registrationformafter
## Test

You can test each mode by changing the configuration of the plugin : 

configrecaptcha

### CODE REVIEW

File: plugins/captcha/recaptcha/recaptcha.php [hunk 1]
```diff
@@ -50,10 +50,11 @@ public function onInit($id = 'dynamic_recaptcha_1')
 			JHtml::_('jquery.framework');
 
 			$theme	= $this->params->get('theme', 'clean');
+			$size	= $this->params->get('size', 'normal');
 			$file	= 'https://www.google.com/recaptcha/api/js/recaptcha_ajax.js';
 
 			JFactory::getDocument()->addScriptDeclaration('jQuery( document ).ready(function()
-				{Recaptcha.create("' . $pubkey . '", "' . $id . '", {theme: "' . $theme . '",' . $this->_getLanguage() . 'tabindex: 0});});');
+				{Recaptcha.create("' . $pubkey . '", "' . $id . '", {theme: "' . $theme . '", size: "' . $size . '",' . $this->_getLanguage() . 'tabindex: 0});});');
```
Comment (andrepereiradasilva) [inline plugins/captcha/recaptcha/recaptcha.php]: Why did you change the version jquery 1.0 code if the option only appears if version 2.0 is selected?

**Comment by LeGfrey** [comment]
Thanks @andrepereiradasilva ! 
My bad :unamused:   I made a mistake checking the IF statement.
Corrected in https://github.com/joomla/joomla-cms/commit/06c4d4b216505b8be8b760cb232409ac4d39c6a2

File: plugins/captcha/recaptcha/recaptcha.php [hunk 2]
```diff
@@ -50,6 +50,7 @@ public function onInit($id = 'dynamic_recaptcha_1')
 			JHtml::_('jquery.framework');
 
 			$theme	= $this->params->get('theme', 'clean');
+			$size	= $this->params->get('size', 'normal');
```
Comment (andrepereiradasilva) [inline plugins/captcha/recaptcha/recaptcha.php]: i think you can remove this too

**Comment by LeGfrey** [comment]
@andrepereiradasilva Sure, thanks :+1: 
 Looks like I'm kind of tired today :tired_face: 
https://github.com/joomla/joomla-cms/commit/2d62d7c627688ab98b78292cdda6b2401060cf7e

**Comment by andrepereiradasilva** [comment]
I have tested this item :white_check_mark: successfully on 2d62d7c627688ab98b78292cdda6b2401060cf7e<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/8909">issues.joomla.org/joomla-cms/8909</a>.</sub>

File: administrator/language/en-GB/en-GB.plg_captcha_recaptcha.ini [hunk 3]
```diff
@@ -29,6 +29,10 @@ PLG_RECAPTCHA_THEME_LIGHT="Light"
 PLG_RECAPTCHA_THEME_DARK="Dark"
 PLG_RECAPTCHA_LANG_LABEL="Language"
 PLG_RECAPTCHA_LANG_DESC="Select the language for the reCAPTCHA. If default is set and the language file has a custom translation, it will be used."
+PLG_RECAPTCHA_SIZE_LABEL="Size"
+PLG_RECAPTCHA_SIZE_DESC="Select the size for the reCAPTCHA field"
```
Comment (brianteeman) [inline administrator/language/en-GB/en-GB.plg_captcha_recaptcha.ini]: Please add a full stop

**Comment by LeGfrey** [comment]
Just added it. Thank you @brianteeman !

**Comment by andrepereiradasilva** [comment]
I have tested this item :white_check_mark: successfully on baf59621b06e1b77e4cc9fbf400d5247ef6e0022<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/8909">issues.joomla.org/joomla-cms/8909</a>.</sub>

**Comment by brianteeman** [comment]
I have tested this item :red_circle: unsuccessfully on baf59621b06e1b77e4cc9fbf400d5247ef6e0022<br /><br />Works perfectly with Captcha 2.0 

But its not implemented for Captcha 1.0

As we currently support both it would be great if it could be applied to Captcha 1.0 as well<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/8909">issues.joomla.org/joomla-cms/8909</a>.</sub>

**Comment by Fedik** [comment]
@brianteeman it have no sense, and it not supported by Captcha 1 https://developers.google.com/recaptcha/old/docs/customization

**Comment by brianteeman** [comment]
Ok no problem then - all good

On 21 January 2016 at 13:18, Fedir Zinchuk <notifications@github.com> wrote:

> @brianteeman <https://github.com/brianteeman> it have no sense, and it
> not supported by Captcha 1
> https://developers.google.com/recaptcha/old/docs/customization
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/joomla/joomla-cms/pull/8909#issuecomment-173565791>.
>



-- 
Brian Teeman
Co-founder Joomla! and OpenSourceMatters Inc.
http://brian.teeman.net/

**Comment by brianteeman** [comment]
Updated my test result based on comment and marked RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/8909">issues.joomla.org/joomla-cms/8909</a>.</sub>
